internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -65,7 +65,7 @@ func (md *machineDeployment) DeployMachinesApp(ctx context.Context) error {
 		// Provide an extra second to try to update the release status.
 		status = "interrupted"
 		var cancel func()
-		ctx, cancel = context.WithTimeout(context.Background(), time.Second)
+		ctx, cancel = context.WithTimeout(ctx, time.Second)
```

Comment:
it's deriving a context from a cancelled context. Not sure that is going to work tbh

Reply:
it does but the derived context's `ctx.Done()` will return a closed channel immediately making any network operation fail (which is the purpose of the extra second)